src/contract/data.rs

```diff
@@ -48,8 +50,8 @@ impl Conceal for Void {
         self.clone()
     }
 }
-impl CommitEncodeWithStrategy for Void {
-    type Strategy = commit_strategy::UsingConceal;
+impl CommitEncode for Void {
+    fn commit_encode<E: io::Write>(self, _e: E) -> usize {0}
```

Comment:
I didn't understand why this was required.

Reply:
It was required because the way it was implemented it was not working. Let me explain.

The code:

```rust
impl CommitEncodeWithStrategy for Void {
    type Strategy = commit_strategy::UsingConceal;
}
```

ultimately will trigger the execution of this other [code](https://github.com/LNP-BP/rust-lnpbp/blob/8e5e1a715efba0704b38fe7d2a24ce71e7b2b23f/client_side_validation/src/client_side_validation.rs#L55-L63):

```rust
impl<T> CommitEncode for amplify::Holder<T, UsingConceal>
where
    T: Conceal,
    <T as Conceal>::Confidential: CommitEncode,
{
    fn commit_encode<E: io::Write>(self, e: E) -> usize {
        self.into_inner().conceal().commit_encode(e)
    }
}

```

If you pay attention, you will notice that this implementation has a constraint that the type returned by the `conceal()` method of the type in question (`Void` in our particular case here) needs to implement the CommitEncode trait.

However, if you look at the implementation of the `conceal()` method for the `Void` structure (shown below), you will notice that it returns the very own type `Void`, which leads to a circular condition.

https://github.com/rgb-org/rgb-core/blob/5984c301c583161643cbecd3ae20405399f43ab4/src/contract/data.rs#L44-L50

So, in the end, the CommitEncode trait is never implemented for the `Void` structure. Not the way it was originally implemented. The proposed change corrects that.

Reply:
If it was never implemented than rust compiler would produce a error – since we are calling this method from the code